CANserver/Logging.cpp

```diff
@@ -290,6 +320,54 @@ void CANServer::Logging::handleMessage(CAN_FRAME *frame, const uint8_t busId)
                     break;
                 }
 
+                case CANServer::Logging::LogType_Drive:
+                {
+                    long currentLogMillis = millis();
+                    int gearNUM = 0;
+                    String tempRow = String(_currentTimeOfDay.tv_sec);
+                    CANServer::CanBus *canbusInstance = CANServer::CanBus::instance();
+                    CANServer::CanBus::AnalysisItemMap::const_iterator found_it = canbusInstance->dynamicAnalysisItems()->find("Gear");
+                    gearNUM = found_it->second->lastValue;
+
+                    if((currentLogMillis - previousDriveLogcycle >= DRIVELOGGING_INTERVAL) && (gearNUM == 2 || gearNUM == 4))
+                    {
+                        previousDriveLogcycle = currentLogMillis;
+                        CANServer::CanBus *canbusInstance = CANServer::CanBus::instance();
+                        for (CANServer::CanBus::AnalysisItemMap::const_iterator iter = canbusInstance->dynamicAnalysisItems()->begin(); iter != canbusInstance->dynamicAnalysisItems()->end(); ++iter)
+                        {
+                            if(iter->second->driveLog == 1){
+                                tempRow = tempRow + "," + String(iter->second->lastValue,6);
+                            }
+                        }
+                        it->second.fileHandle.println(tempRow);
+                    }
+                    break;
+                }
+
+                case CANServer::Logging::LogType_Charge:
+                {
+                    long currentLogMillis = millis();
+                    int chargeStatus = 0;
+                    String tempRow = String(_currentTimeOfDay.tv_sec);
+                    CANServer::CanBus *canbusInstance = CANServer::CanBus::instance();
+                    CANServer::CanBus::AnalysisItemMap::const_iterator found_it = canbusInstance->dynamicAnalysisItems()->find("UIChargeStatus");
+                    chargeStatus = found_it->second->lastValue;
```

Comment:
Similar to the drive log check.  Make sure that the chargeStatus analysis item exists before using it